docs/reference/sql/functions/aggs.asciidoc

```diff
@@ -53,11 +53,42 @@ COUNT(expression<1>)
 
 Returns the total number (count) of input values.
 
+In case of `COUNT(*)` or `COUNT(<numeric_value>)`, _all_ values are considered (including `null` or missing ones).
+
+In case of `COUNT(<field_name>)` `null` values are not considered.
+
+
 ["source","sql",subs="attributes,macros"]
 --------------------------------------------------
 include-tagged::{sql-specs}/docs.csv-spec[aggCountStar]
 --------------------------------------------------
 
+
+[[sql-functions-aggs-count-all]]
+===== `COUNT(ALL)`
+
+.Synopsis:
+[source, sql]
+--------------------------------------------------
+COUNT(ALL field_name<1>)
```

Comment:
since `ALL` is optional maybe wrap it in `[]`

Reply:
There are 3 separate docs sections for `COUNT`:
* `COUNT`
* `COUNT(ALL)`
* `COUNT(DISTINCT)`

and each has its own description and examples. If I would have treated `COUNT` and `COUNT(ALL)` in the same section I would have used the square brackets (and the first version of the documentation I had did it like this in fact), but if it has a its own separate section, I would leave it as is now.